### PR TITLE
Replace Condvars with blocking channel

### PR DESCRIPTION
**Pull request by NikVolf**
Should've been done it initially.

### CODE REVIEW

File: primitives/io/src/batch_verifier.rs [hunk 1]
```diff
@@ -139,14 +137,12 @@ impl BatchVerifier {
 		self.sr25519_items.clear();
 
 		if pending.len() > 0 {
-			let pair = Arc::new((Mutex::new(false), Condvar::new()));
-			let pair_clone = pair.clone();
-
+			let (sender, receiver) = std::sync::mpsc::channel();
```
Comment (bkchr) [inline primitives/io/src/batch_verifier.rs:140]: I still don't understand why all this code isn't just:
```
futures::executor::block_on(futures::future::join_all(pending))
```

File: primitives/io/src/batch_verifier.rs [hunk 2]
```diff
@@ -155,12 +151,8 @@ impl BatchVerifier {
...
 				return false;
 			}
-
-			let (finished, cond_var) = &*pair;
-			let mut finished = finished.lock().expect("Locking can only fail when the mutex is poisoned; qed");
-			while !*finished {
-				finished = cond_var.wait(finished).expect("Waiting can only fail when the mutex waited on is poisoned; qed");
-			}
+			receiver.recv()
+				.expect("This can panic only if the spawned tasks is panicked. Spawned task does not panic; qed.");
```
Comment (bkchr) [inline primitives/io/src/batch_verifier.rs:155]: This is a claim that you can not verify. Either we ignore the error or print some warning on error.

File: primitives/io/src/batch_verifier.rs (hunk 1 above)
Comment (NikVolf) [reply primitives/io/src/batch_verifier.rs:140]: Integration tests (`test_sync`) are panicking, they use it and you can't have one executor inside another.

**Comment by bkchr** [comment]
Fails to compile

File: primitives/io/src/batch_verifier.rs [hunk 3]
```diff
@@ -155,11 +151,9 @@ impl BatchVerifier {
 
 				return false;
 			}
-
-			let (finished, cond_var) = &*pair;
-			let mut finished = finished.lock().expect("Locking can only fail when the mutex is poisoned; qed");
-			while !*finished {
-				finished = cond_var.wait(finished).expect("Waiting can only fail when the mutex waited on is poisoned; qed");
+			if let Err(e) = receiver.recv() {
+				log::warn!(target: "runtime", "Haven't received async result from verification task. Returning false.");
```
Comment (bkchr) [inline primitives/io/src/batch_verifier.rs:155]:
```suggestion
			if receiver.recv().is_err() {
				log::warn!(target: "runtime", "Haven't received async result from verification task. Returning false.");
```

**Pull request merged**